### ISSUE TITLE
How to change to only LSTM without CRF

### ISSUE THREAD

**Issue by funqc** [SkyAndCloud/bilstm_crf_sequence_labeling_pytorch#1]
How should I calculate the loss with the mask?

**Comment by funqc** [comment]
```python
    def neg_log_likelihood(self, seq, tags, mask):
        feats = self.get_lstm_features(seq, mask)
        seq_len, batch_size, tag_size = feats.size()
        tag_scores = F.log_softmax(feats, dim=1) * mask.unsqueeze(-1)
        loss = feats.new_zeros(seq_len)
        loss_function = nn.NLLLoss()
        for t, tag_score in enumerate(tag_scores):
            loss[t] = loss_function(tag_score, tags[t])
        return loss.sum()
```
It's that right?

**Comment by RunhangZhang** [comment]
I also want to know this, is that right? And how about predict? How to return best_path?

**Comment by 19890843006** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
airon90, @chamow97, @bonfy, @rd10397, @developer-don, @filiperocha, @dongphuchaitrieu, @einaralex, @zkilgore, @chriswburke, @punchagan, @benji13, @SarthakS93, @kindy, @athlx, @marcusmonteiro, @shovelrider, @William0Friend, @parallelExecution, @tuteng

**Comment by Ererere911** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online). Hurry, applications close on [March 1, 2024](https://jobsindeed.online).

## Job Description:
- **Position:** Senior Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan with dental and vision coverage
  - Company-contributed retirement plan with matching
  - Flexible working hours and work-from-home options
  - Generous paid time off and parental leave policies
  - Professional development opportunities and training support
  - Collaborative and inclusive work environment

## Requirements:
- Minimum of 5 years of experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Extensive experience with frameworks such as React, Angular, or Vue.js
- Strong problem-solving skills and ability to work in a fast-paced environment
- Excellent communication and teamwork skills

**Note:** This position is open to candidates from anywhere in the world. Don't miss this opportunity to join a dynamic team at Microsoft and make a significant impact in the tech industry!
thyrlian, @arthuroy, @therod, @mathieugagne, @DvJ4Y, @kevinhaas, @khanhtdbse, @paranoia445, @pp-sitt, @mfuentesg, @rcdosado, @jakeprem, @desireco, @peebeejay, @comunidadconocimiento, @gonzar11, @niuguy, @IamRaviTejaG-zz, @nsanthoshkumar, @afrizaloky
